### PR TITLE
shell completion: support external commands

### PR DESCRIPTION
**Pull request by xu-cheng**
I only test this under zsh. Would be nicer if someone tests it for bash as well. Thanks. 
cc @Homebrew/owners

### CODE REVIEW

**Comment by bfontaine** [comment]
Seems to be broken for Bash. Without these changes if I press <kbd>Tab</kbd> after `brew` it’ll show me all the possibilities. With these it only gives me `--cache`.

File: Library/Contributions/brew_bash_completion.sh [hunk 1]
```diff
@@ -580,62 +580,7 @@ _brew ()
...
-            unlink
-            unlinkapps
-            unpack
-            unpin
-            untap
-            update
-            upgrade
-            uses
-            "
+        __brewcomp $(brew commands --quiet --include-aliases)
```
Comment (bfontaine) [inline Library/Contributions/brew_bash_completion.sh]: Add quotes around `$(…)`:

```sh
__brewcomp "$(brew commands --quiet --include-aliases)"
```

Comment (xu-cheng) [inline Library/Contributions/brew_bash_completion.sh]: Fixed, Thanks @bfontaine

**Comment by xu-cheng** [comment]
Any other comments from other maintainers?

**Comment by mikemcquaid** [comment]
If it works in both: :+1:

**Comment by xu-cheng** [comment]
It works, thought I think you may have comments on new `brew commands` flags and possible manpage change.

**Comment by DomT4** [comment]
As long as they work, pretty much always :+1: on expanding the completions. The new flags look fine.

**Pull request closed without merging**